### PR TITLE
CMS Extension 

### PR DESCRIPTION
**Pull request by abist**
The CMS Extension. Need to bump toolsservice version once CI queues are done and the PR there is merged in (https://github.com/Microsoft/sqltoolsservice/pull/782)

### CODE REVIEW

File: scripts/sql-test-integration.bat [hunk 1]
```diff
@@ -8,7 +8,7 @@ echo %VSCODEUSERDATADIR%
 echo %VSCODEEXTENSIONSDIR%
 @echo OFF
 
-call .\scripts\code.bat --extensionDevelopmentPath=%~dp0\..\extensions\integration-tests --extensionTestsPath=%~dp0\..\extensions\integration-tests\out --user-data-dir=%VSCODEUSERDATADIR% --extensions-dir=%VSCODEEXTENSIONSDIR% --remote-debugging-port=9222
+call .\scripts\code.bat --extensionDevelopmentPath=%~dp0\..\extensions\cms\src\test --extensionTestsPath=%~dp0\..\extensions\cms\out\test --user-data-dir=%VSCODEUSERDATADIR% --extensions-dir=%VSCODEEXTENSIONSDIR% --remote-debugging-port=9222
```
Comment (anthonydresser) [inline scripts/sql-test-integration.bat]: I'm assuming you don't want to check this in

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -3776,6 +3776,11 @@ declare module 'azdata' {
 		errorCode: number;
 	}
 
+	export enum CmsDialog {
```
Comment (anthonydresser) [inline src/sql/azdata.proposed.d.ts]: This is unfortunate. Ideally this extension should require no or minimal changes to core code and certainly shouldn't have any references to cms directly. Can you post a screenshot of the dialog and a comment on why these core changes are necessary?

Comment (abist) [reply src/sql/azdata.proposed.d.ts]: The connection dialog behaves differently when prompted for a connection via the CMS dialog. We don't have a "generalized" connection dialog so I had to add this additional parameter.

Comment (anthonydresser) [reply src/sql/azdata.proposed.d.ts]: @kburtram @alanrenmsft thoughts? I know this is one of the apis we were trying to stabilize.

**Comment by abist** [comment]
Closing this PR since I have a new one out taking care of the comments here.

**Pull request closed without merging**